packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -749,4 +751,13 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 
 		return executions.map(({ id }) => id);
 	}
+
+	async allCurrentExecutionIds() {
+		const executions = await this.find({
+			select: ['id'],
+			where: { status: In(['new', 'running']) },
+		});
+
+		return executions.map(({ id }) => id);
+	}
```

Comment:
maybe we can update this to look only for executions that started over an hour ago or so? 
checking for all executions could be expensive for instances that see a lot of executions.

Reply:
[f0fc745](https://github.com/n8n-io/n8n/pull/9676/commits/f0fc745af01bd556f766eb49cb1f8c08555f1c13)